replications/transport/http_test.go

```diff
@@ -0,0 +1,352 @@
+package transport
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/kit/feature"
+	"github.com/influxdata/influxdb/v2/kit/platform"
+	"github.com/influxdata/influxdb/v2/mock"
+	"github.com/stretchr/testify/assert"
+	tmock "github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+)
+
+var (
+	orgStr            = "1234123412341234"
+	orgID, _          = platform.IDFromString(orgStr)
+	remoteStr         = "9876987698769876"
+	remoteID, _       = platform.IDFromString(remoteStr)
+	idStr             = "4321432143214321"
+	id, _             = platform.IDFromString(idStr)
+	localBucketStr    = "1111111111111111"
+	localBucketId, _  = platform.IDFromString(localBucketStr)
+	remoteBucketStr   = "1234567887654321"
+	remoteBucketID, _ = platform.IDFromString(remoteBucketStr)
+	testReplication   = influxdb.Replication{
+		ID:                *id,
+		OrgID:             *orgID,
+		RemoteID:          *remoteID,
+		LocalBucketID:     *localBucketId,
+		RemoteBucketID:    *remoteBucketID,
+		Name:              "example",
+		MaxQueueSizeBytes: influxdb.DefaultReplicationMaxQueueSizeBytes,
+	}
+)
+
+func TestReplicationHandler(t *testing.T) {
```

Comment:
Could probably run this and/or the subtests in parallel

Reply:
I get worried about parallelizing the HTTP tests because (if I'm reading the `httptest` code correctly) they actually bind a port per-server instance, so blasting too many tests at once could create contention there.